### PR TITLE
Law and test for LJ potential

### PR DESCRIPTION
**Pull request by alesanter**

### CODE REVIEW

File: symplyphysics/laws/chemistry/lennard_jones_potential.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+from sympy import Eq
```
Comment (blackyblack) [inline symplyphysics/laws/chemistry/lennard_jones_potential.py:1]: Don't use marketing names for laws.

File: symplyphysics/laws/chemistry/lennard_jones_potential.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+## other at a very close distance, attract each other at moderate distance, and do not interact at infinite
+## distance.
+
+# Law: U = 4*e*((sigma/r)**12 - (sigma/r)**6)
+## U - Lennard-Jones potential
+## e - depth of potential well ("dispersion energy")
+## sigma - distance at which the potential energy is zero ("particle size")
+## r - distance between interacting particles
+
+lj_potential = Symbol("potential", units.energy)
```
Comment (blackyblack) [inline symplyphysics/laws/chemistry/lennard_jones_potential.py:24]: Don't shortcut and use descriptive name.

File: test/chemistry/lennard_jones_potential_test.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+# Description
+## The value of the Lennard-Jones potential for two interacting Xenon atoms at the distance of
+## 1 Å is 2.65e-13 J (e = 2.94e-21 J, sigma = 4.1 Å)
+
+
+@fixture(name="test_args")
+def test_args_fixture():
+    e = Quantity(2.94e-21 * units.joule)
+    sigma = Quantity(4.10e-10 * units.angstrom)
```
Comment (blackyblack) [inline test/chemistry/lennard_jones_potential_test.py:20]: Number does not match with description.

File: symplyphysics/laws/chemistry/lennard_jones_potential.py (hunk 1 above)
Comment (alesanter) [reply symplyphysics/laws/chemistry/lennard_jones_potential.py:1]: There are lots of other potentials that target the same topic (Mie, Buckingham, Morse potentials) though

Comment (alesanter) [reply symplyphysics/laws/chemistry/lennard_jones_potential.py:1]: Maybe `lennard_jones_intermolecular_pair_potential`?

Comment (blackyblack) [reply symplyphysics/laws/chemistry/lennard_jones_potential.py:1]: Hm, good point. I suggest to add folder, eg `potential_energy_models` and put different models in it. This way we can use fancy model names without cluttering the project.

File: test/chemistry/lennard_jones_potential_test.py [hunk 4]
```diff
@@ -17,7 +17,7 @@
 @fixture(name="test_args")
 def test_args_fixture():
     e = Quantity(2.94e-21 * units.joule)
-    sigma = Quantity(4.10e-10 * units.angstrom)
+    sigma = Quantity(4.10 * units.angstrom)
```
Comment (blackyblack) [inline test/chemistry/lennard_jones_potential_test.py:20]: How is it possible, that 10 orders of magnitude didn't change the test results?

File: symplyphysics/laws/chemistry/lennard_jones_potential.py (hunk 1 above)
Comment (alesanter) [reply symplyphysics/laws/chemistry/lennard_jones_potential.py:1]: I agree

File: test/chemistry/lennard_jones_potential_test.py (hunk 4 above)
Comment (alesanter) [reply test/chemistry/lennard_jones_potential_test.py]: I'm surprised myself. I'll look into it

File: symplyphysics/laws/chemistry/lennard_jones_potential.py (hunk 1 above)
Comment (blackyblack) [reply symplyphysics/laws/chemistry/lennard_jones_potential.py]: I think folder should have `__init__.py` to work properly. Tests don't need it.

File: symplyphysics/laws/chemistry/potential_energy_models/lennard_jones_potential.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+## sigma - distance at which the potential energy is zero ("particle size")
+## r - distance between interacting particles
+
+potential = Symbol("potential", units.energy)
+dispersion_energy = Symbol("dispersion_energy", units.energy)
+particle_size = Symbol("particle_size", units.length)
+distance = Symbol("distance", units.length)
+reduced_distance = distance / particle_size
+
+law = Eq(potential, 4 * dispersion_energy * (reduced_distance**(-12) - reduced_distance**(-6)))
```
Comment (blackyblack) [inline symplyphysics/laws/chemistry/potential_energy_models/lennard_jones_potential.py:30]: Why did you invert the formula? Now it does not look like the law in description.

File: test/chemistry/lennard_jones_potential_test.py (hunk 4 above)
Comment (alesanter) [reply test/chemistry/lennard_jones_potential_test.py]: Perhaps, this is the behaviour of `approx`. With the correct `sigma` value of `4.10 Å`, the potential value is `2.65e-13 J` as expected, and with the wrong value of `4.10e-10 Å`, the potential value calculated is `-5.59e-77 J`. The documentation for `approx` says that it uses either `math.isclose` or `numpy.isclose` under the hood. Although only `numpy` gives `True`
image

Comment (blackyblack) [reply test/chemistry/lennard_jones_potential_test.py]: Use it like this:
`assert result_value == approx(2.65e-13, abs=1e-15)`
For some reason, tolerance is overriden by some default value, unless it is absolute.

Comment (alesanter) [reply test/chemistry/lennard_jones_potential_test.py]: I also came to that conclusion

File: symplyphysics/laws/chemistry/potential_energy_models/lennard_jones_potential.py (hunk 5 above)
Comment (blackyblack) [reply symplyphysics/laws/chemistry/potential_energy_models/lennard_jones_potential.py:30]: Simply remove reduced distance and write law in the same form as in description.

**Review by blackyblack — APPROVED** [review]

**Pull request merged**